### PR TITLE
Restructure workflow docs

### PR DESCRIPTION
**Pull request by Noah-Silvera**
This is a restructure of the workflow docs to bring them in line with the organizational pattern set by the web documentation.

- Use cases have been turned into tutorials and moved into their respective SDK sections
- Each tutorial should now be a self contained, complete example.
- Activity / Form reference articles have been added where appropiate
- An absolute basics "Create Activity" article has been added to the start of the SDK docs for each section
- Various improvements and cleanup

### CODE REVIEW

File: docs/workflow/tutorial-web-add-layer-to-map.mdx [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+title: Add a Layer to a Map with an Activity
+description: Geocortex Workflow - Learn how to use special activity properties to access and mutate the application map
+---
+
+import WebPrereqs from "./snippets/prereqs-web.mdx";
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+Some activities may need access to resources like the map, the host application or host widget. The `AppActivity` base class can be used to extend an existing custom activity to access these resources.
```
Comment (kenlyon) [inline docs/workflow/tutorial-web-add-layer-to-map.mdx:9]:
```suggestion
Some activities may need access to resources like the map, the host application or host widget. The `AppActivity` base class can be used to extend an existing activity to access these resources.
```

File: sidebars.js [hunk 2]
```diff
@@ -124,60 +124,61 @@ module.exports = {
         "workflow/concepts-key-concepts",
         {
             type: "category",
-            label: "Use Cases",
+            label: "Geocortex Workflow JavaScript SDK",
```
Comment (kenlyon) [inline sidebars.js:127]:
```suggestion
            label: "Geocortex Workflow TypeScript SDK",
```

File: docs/workflow/sdk-net-register-activities.mdx [hunk 3]
```diff
@@ -3,31 +3,63 @@ title: Registering .NET Activities with Geocortex Workflow Designer
...
+```
 
 This activity will need a TypeScript stub that defines its input and output types and descriptions, specifies the runtime environments it supports, and an activity description.
 
 ### Create a new Activity Pack
 
-Follow the instructions in [Implement a Custom Activity for Web Applications](usecases-web-implement-custom-activity.mdx) up to [Implement the Activity](usecases-web-implement-custom-activity.mdx#implement-the-activity).
+Follow the instructions in [Implement a Custom Activity for Web Applications](tutorial-web-calculate-logarithm-activity.mdx) up to [Implement the Activity](tutorial-web-calculate-logarithm-activity.mdx#implement-the-activity).
 
 ### Stub out the Activity Metadata
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/workflow/sdk-net-register-activities.mdx:65]: That's a good idea!

I tried importing the `ActivityNotImplementedError`, but it doesn't seem to be exported as part of the workflow SDK, unless I'm importing it wrong

```ts
import { ActivityNotImplementedError } from "@geocortex/workflow/runtime/Errors";


export interface CalculateLogInputs {
    // @description The base of the logarithm. Defaults to the natural log.
    base?: number;

    // @description The number to calculate the logarithm for.
    // @required
    value: number;
}

export interface CalculateLogOutputs {
    // @description The logarithm output.
    result: number;
}

// @supportedApps GMV
// @category Custom Activities
// @description An activity that calculates the log of a number with the given base.
export class CalculateLog {
    static action = "your:unique:namespace::CalculateLog";

    static suite = "your:unique:namespace";

    async execute(inputs: CalculateLogInputs): Promise<CalculateLogOutputs> {
        // The host application must implement this activity
        throw new ActivityNotImplementedError(CalculateLog.action);
    }
}
```

How would you feel about throwing a Generic error instead? e.g.

```ts
throw new Error("Activity not implemented for this platform")
```

[the remark this answers is not in the document]
Comment (kenlyon) [reply docs/workflow/sdk-net-register-activities.mdx:65]: I think that would work for now. Maybe at some point we can make `ActivityNotImplementedError` more visible but it certainly addresses the immediate concern.

File: docs/workflow/sdk-server-create-activity.mdx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+---
+
+import ServerPrereqs from "./snippets/prereqs-server.mdx";
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+This article will walk you through creating a new workflow activity for Geocortex Workflow Server.
+
+## Prerequisites
+
+<ServerPrereqs />
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/workflow/sdk-server-create-activity.mdx:13]: right... thanks for the reminder! did a find and replace throughout the docs

File: docs/workflow/sdk-net-overview.mdx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+sidebar_label: Overview
+description: Geocortex Workflow - Overview of the .NET workflow SDK
+---
+
+import UseCaseContainer from "../../src/components/UseCaseContainer";
+import UseCaseCard from "../../src/components/UseCaseCard";
+
+Geocortex Workflow supports has a .NET runtime for executing workflows in a .NET environment. [Geocortex Mobile](https://docs.geocortex.com/mobileviewer/) and [Geocortex Workflow Server](https://docs.geocortex.com/workflow/latest/help/Default.htm#wf5/help/server-workflows.htm) both run workflows in a .NET environment.
+
+Workflows for Geocortex Mobile and Geocortex Workflow Server are still authored in the [Geocortex Workflow Designer](https://apps.geocortex.com/workflow/designer/). Custom .NET workflow activities can be executed by name with the [Run Activity](https://docs.geocortex.com/workflow/latest/help/Default.htm#wf5/help/activities/run-activity.htm) activity. In addition, the Geocortex Workflow JavaScript SDK can be used to create [activity stubs](sdk-net-register-activities.mdx) that give a user friendly interface for custom .NET workflow activities.
```
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/workflow/sdk-net-overview.mdx]: Totally. Updated.

**Comment by Noah-Silvera** [comment]
> Should the renaming to "Geocortex Workflow Designer" be done across the board. I see the following places that still just call it "Workflow Designer":
> 
> * docs\mobile\usecases-change-default-map-click-behavior.mdx
> * docs\web\tutorial-change-default-map-click-behavior.mdx
> * docs\workflow\sdk-mobile-create-activity.mdx
> * docs\workflow\sdk-server-create-activity.mdx
> * docs\workflow\sdk-web-activity-block-tags.mdx
> * docs\workflow\sdk-web-overview.mdx
> * docs\workflow\tutorial-mobile-activity-indicator-form-element.mdx
> * docs\workflow\tutorial-mobile-calculate-logarithm-activity.mdx
> * docs\workflow\tutorial-mobile-show-map-callout.mdx
> * docs\workflow\tutorial-server-calculate-logarithm-activity.mdx
> * docs\workflow\tutorial-web-add-layer-to-map.mdx
> * docs\workflow\tutorial-web-calculate-logarithm-activity.mdx
> * docs\workflow\tutorial-web-implement-star-rating-form-element.mdx

Yes, it should be done across the board. 
Updated.

File: docs/workflow/sdk-net-register-activities.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Noah-Silvera) [reply docs/workflow/sdk-net-register-activities.mdx:65]: Done!

**Comment by kenlyon** [comment]
> 
> 
> > Should the renaming to "Geocortex Workflow Designer" be done across the board. I see the following places that still just call it "Workflow Designer":
> > 
> > * docs\mobile\usecases-change-default-map-click-behavior.mdx
> > * docs\web\tutorial-change-default-map-click-behavior.mdx
> > * docs\workflow\sdk-mobile-create-activity.mdx
> > * docs\workflow\sdk-server-create-activity.mdx
> > * docs\workflow\sdk-web-activity-block-tags.mdx
> > * docs\workflow\sdk-web-overview.mdx
> > * docs\workflow\tutorial-mobile-activity-indicator-form-element.mdx
> > * docs\workflow\tutorial-mobile-calculate-logarithm-activity.mdx
> > * docs\workflow\tutorial-mobile-show-map-callout.mdx
> > * docs\workflow\tutorial-server-calculate-logarithm-activity.mdx
> > * docs\workflow\tutorial-web-add-layer-to-map.mdx
> > * docs\workflow\tutorial-web-calculate-logarithm-activity.mdx
> > * docs\workflow\tutorial-web-implement-star-rating-form-element.mdx
> 
> Yes, it should be done across the board.
> Updated.

I see two outstanding offenders:
1. docs\mobile\tutorial-run-workflow-app-config.mdx:149 says "import it into workflow designer."
1. docs\workflow\sdk-web-activity-block-tags.mdx:6 has a hyperlink with link text of "Workflow Designer" instead of "Geocortex Workflow Designer".

**Comment by Noah-Silvera** [comment]
> I see two outstanding offenders:
> 
> 1. docs\mobile\tutorial-run-workflow-app-config.mdx:149 says "import it into workflow designer."
> 2. docs\workflow\sdk-web-activity-block-tags.mdx:6 has a hyperlink with link text of "Workflow Designer" instead of "Geocortex Workflow Designer".

Keen attention to detail as always Ken! Thanks.

Fixed.

**Pull request merged**